terraform/generate-iam.tf

```diff
@@ -0,0 +1,211 @@
+# AWS Batch role and policy
+resource "aws_iam_role" "aws_batch_service_role" {
+  name = "${var.prefix}-batch-service-role"
+  assume_role_policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Effect" : "Allow",
+        "Principal" : {
+          "Service" : "batch.amazonaws.com"
+        },
+        "Action" : "sts:AssumeRole"
+      }
+    ]
+  })
+  permissions_boundary = "arn:aws:iam::${local.account_id}:policy/NGAPShRoleBoundary"
+  tags                 = local.default_tags
+}
+
+resource "aws_iam_role_policy_attachment" "aws_batch_service_role_policy_attach" {
+  role       = aws_iam_role.aws_batch_service_role.name
+  policy_arn = aws_iam_policy.batch_service_role_policy.arn
+}
+
+resource "aws_iam_policy" "batch_service_role_policy" {
+  name        = "${var.prefix}-batch-service-role-policy"
+  description = "Provides access for the AWS Batch service to manage the required resources, including Amazon EC2 and Amazon ECS resources"
+  policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Effect" : "Allow",
+        "Action" : [
+          "ec2:DescribeAccountAttributes",
+          "ec2:DescribeInstances",
+          "ec2:DescribeInstanceAttribute",
+          "ec2:DescribeSubnets",
+          "ec2:DescribeSecurityGroups",
+          "ec2:DescribeKeyPairs",
+          "ec2:DescribeImages",
+          "ec2:DescribeImageAttribute",
+          "ec2:DescribeInstanceStatus",
+          "ec2:DescribeSpotInstanceRequests",
```

Comment:
I know it's annoying finding the least privileges necessary but I think we could at least get rid of the *Spot* permissions because we aren't going to be doing anything with spot instances at this time

Reply:
I will remove all references to Spot permissions since we will not be using Spot instances in our infrastructure.